crates/client/src/client.rs

```diff
@@ -240,6 +240,28 @@ impl FuelClient {
         Self::decode_response(response)
     }
 
+    pub async fn query_with_headers<ResponseData, Vars>(
+        &self,
+        q: Operation<ResponseData, Vars>,
+        headers: impl IntoIterator<Item = (String, String)>,
```

Comment:
TODO: HeadersMap here?

Reply:
I assume this one https://docs.rs/http/1.2.0/http/header/struct.HeaderMap.html - makes sense :+1:

Reply:
Forgot to mention this is done, see d9091e6465f140db830bd5f626dc4535a8ba93d0